### PR TITLE
TensorFlow Graph for Style Transfer

### PR DESCRIPTION
**Pull request by shreyajain17**

### CODE REVIEW

File: src/python/turicreate/toolkits/style_transfer/_tf_model_architecture.py [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+
+        for key in net_params.keys():
+            net_params[key] = _utils.convert_shared_float_array_to_numpy(net_params[key])
+
+        for key in config.keys():
+            config[key] = _utils.convert_shared_float_array_to_numpy(config[key])
+
+        self._batch_size = 1
+        self._finetune_all_params = True
+        self._define_training_graph = True
```
Comment (abhishekpratapa) [inline src/python/turicreate/toolkits/style_transfer/_tf_model_architecture.py]: config `st_training` true

File: src/python/turicreate/toolkits/_tf_utils.py [hunk 2]
```diff
@@ -58,6 +58,26 @@ def convert_conv1d_coreml_to_tf(conv_weights):
...
+    conv_weights: 4d numpy array of shape 
+       [outputChannels, kernelChannels, kernelHeight, kernelWidth] 
+
+    Returns
+    -------
+	return: 4d numpy array of shape
+	   [kernelWidth, kernelHeight, kernelChannels, outputChannels]
+
+	"""
+	conv_weights = np.transpose(conv_weights, (3, 2, 0, 1))
```
Comment (nickjong) [inline src/python/turicreate/toolkits/_tf_utils.py]: If the input is OIHW, doesn't a `(3, 2, 0, 1)` transpose yield WHOI, not WHIO?

And isn't the TF format HWIO? https://www.tensorflow.org/api_docs/python/tf/nn/conv2d

File: src/python/turicreate/toolkits/_tf_utils.py [hunk 3]
```diff
@@ -150,6 +170,24 @@ def convert_conv1d_tf_to_coreml(conv_weights):
...
+    conv_weights: 4d numpy array of shape
+	   [kernelWidth, kernelHeight, kernelChannels, outputChannels]
+
+    Returns
+    -------
+	return: 4d numpy array of shape 
+       [outputChannels, kernelChannels, kernelHeight, kernelWidth] 
+
+	"""
+	return conv_weights
```
Comment (nickjong) [inline src/python/turicreate/toolkits/_tf_utils.py]: This doesn't appear to do anything

File: src/python/turicreate/toolkits/_tf_utils.py [hunk 4]
```diff
@@ -58,6 +58,26 @@ def convert_conv1d_coreml_to_tf(conv_weights):
...
+
+	Parameters
+    ----------
+    conv_weights: 4d numpy array of shape 
+       [outputChannels, kernelChannels, kernelHeight, kernelWidth] 
+
+    Returns
+    -------
+	return: 4d numpy array of shape
+	   [kernelWidth, kernelHeight, kernelChannels, outputChannels]
```
Comment (nickjong) [inline src/python/turicreate/toolkits/_tf_utils.py]: This comment doesn't match the implementation below

File: src/python/turicreate/toolkits/_tf_utils.py [hunk 5]
```diff
@@ -150,6 +170,25 @@ def convert_conv1d_tf_to_coreml(conv_weights):
...
 
+def convert_conv2d_tf_to_coreml(conv_weights):
+	"""
+	Convolutional weights from TensorFlow in the format [kernelWidth, kernelChannels, outputChannels]
+	are converted back in CoreML specifications [outputChannels, kernelChannels, kernelHeight, kernelWidth].
+
+	Parameters
+    ----------
+    conv_weights: 4d numpy array of shape
+	   [kernelWidth, kernelHeight, kernelChannels, outputChannels]
```
Comment (nickjong) [inline src/python/turicreate/toolkits/_tf_utils.py]: This comment doesn't match the implementation below

File: src/python/turicreate/toolkits/style_transfer/_tf_model_architecture.py [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+    Returns
+    -------
+    out: dict
+        The TF Variable dictionary.
+    """
+    tensorflow_variables = dict()
+    for key, value in net_params.items():
+        if 'weight' in key:
+            if 'conv' in key:
+                tensorflow_variables[key] = _tf.Variable(initial_value = _utils.convert_conv2d_tf_to_coreml(net_params[key]), name=key)x
```
Comment (nickjong) [inline src/python/turicreate/toolkits/style_transfer/_tf_model_architecture.py]: Is this trailing `x` a typo? Is this tested?

**Pull request merged**